include/libopencm3/swm050/gpio.h

```diff
@@ -47,52 +44,74 @@
 #define GPIO7				(1 << 7)
 #define GPIO8				(1 << 8)
 #define GPIO9				(1 << 9)
-#define GPIO_ALL			0x3ff
-/**@}*/
+#define GPIO_ALL			(0x3ff)
+/*@}*/
 
 /* GPIO direction definitions */
 /** @defgroup gpio_dir GPIO Pin Direction
 @{*/
-#define GPIO_INPUT			0x0
-#define GPIO_OUTPUT			0x1
-/**@}*/
+#define GPIO_INPUT			(0x0)
+#define GPIO_OUTPUT			(0x1)
+/*@}*/
+
+/* GPIO polarity definitions */
+/** @defgroup gpio_pol GPIO Polarity
+@{*/
+#define GPIO_LOW			(0x0)
+#define GPIO_HIGH			(0x1)
+/*@}*/
 
+/* GPIO interrupt trigger definitions */
+/** @defgroup gpio_trig_type GPIO Interrupt Trigger Type
+@{*/
+#define GPIO_LEVEL			(0x0)
+#define GPIO_EDGE			(0x1)
+/*@}*/
+
+/* GPIO interrupt mask definitions */
+/** @defgroup gpio_int_masked GPIO Interrupt Mask
+@{*/
+#define GPIO_UNMASKED			(0x0)
+#define GPIO_MASKED			(0x1)
+/*@}*/
+
+/* GPIO Registers */
 /** @defgroup gpio_registers GPIO Registers
 @{*/
 /** Data register */
-#define GPIO_DATA			MMIO32(GPIO_BASE + 0x0)
+#define ADATA				MMIO32(GPIO_BASE + 0x0)
 /** Direction register */
-#define GPIO_DIR			MMIO32(GPIO_BASE + 0x4)
+#define ADIR				MMIO32(GPIO_BASE + 0x4)
 /** Interrupt enable register */
-#define GPIO_INTEN			MMIO32(GPIO_BASE + 0x30)
+#define INTEN_A				MMIO32(GPIO_BASE + 0x30)
 /** Interrupt mask register */
-#define GPIO_INTMASK			MMIO32(GPIO_BASE + 0x34)
+#define INTMASK_A			MMIO32(GPIO_BASE + 0x34)
 /** Interrupt trigger mode register */
-#define GPIO_INTLEVEL			MMIO32(GPIO_BASE + 0x38)
+#define INTLEVEL_A			MMIO32(GPIO_BASE + 0x38)
 /** Interrupt polarity register */
-#define GPIO_INTPOLARITY		MMIO32(GPIO_BASE + 0x3c)
+#define INTPOLARITY_A			MMIO32(GPIO_BASE + 0x3c)
 /** Interrupt status after masking */
-#define GPIO_INTSTATUS			MMIO32(GPIO_BASE + 0x40)
+#define INTSTAT_A			MMIO32(GPIO_BASE + 0x40)
 /** Interrupt status before masking */
-#define GPIO_INTRAWSTATUS		MMIO32(GPIO_BASE + 0x44)
+#define RAWINTSTAT_A			MMIO32(GPIO_BASE + 0x44)
 /** Interrupt clear register */
-#define GPIO_INTEOI			MMIO32(GPIO_BASE + 0x48)
+#define INTEOI_A			MMIO32(GPIO_BASE + 0x48)
 /** External register (wat) */
-#define GPIO_EXT			MMIO32(GPIO_BASE + 0x4c)
-
-/**@}*/
+#define AEXT				MMIO32(GPIO_BASE + 0x4c)
+/*@}*/
 
+/* SYSCON Registers */
 /** @defgroup syscon_register SYSCON Registers
  * @note These registers are really part of the SYSCON system control space
  * @{*/
 /** SWD Enable register */
 #define SWD_SEL				MMIO32(SYSTEM_CON_BASE + 0x30)
-/** GPIO Alternat function selection register */
-#define GPIO_SEL			MMIO32(SYSTEM_CON_BASE + 0x80)
+/** GPIO Alternate function selection register */
+#define PORTA_SEL			MMIO32(SYSTEM_CON_BASE + 0x80)
 /** GPIO Pull up register */
-#define GPIO_PULLUP			MMIO32(SYSTEM_CON_BASE + 0x90)
+#define PORTA_PULLUP			MMIO32(SYSTEM_CON_BASE + 0x90)
 /** GPIO Input enable register */
-#define GPIO_INEN			MMIO32(SYSTEM_CON_BASE + 0xe0)
+#define PORTA_INEN			MMIO32(SYSTEM_CON_BASE + 0xe0)
```

Comment:
these could potentially be changed to GPIO_PORTA_XXXX but I'd probably say to just leave them as is for now.